htdocs/lib2/logic/user.class.php

```diff
@@ -1483,16 +1483,7 @@ public function canDelete()
         global $login;
         $login->verify();
 
-        if ($login->userid != $this->nUserId && ($login->admin & ADMIN_USER) != ADMIN_USER) {
-            return false;
-        }
-
-        return
-            sql_value("SELECT COUNT(*) FROM `caches` WHERE `user_id`='&1'", 0, $this->nUserId)
-            + sql_value("SELECT COUNT(*) FROM `cache_logs` WHERE `user_id`='&1'", 0, $this->nUserId)
-            + sql_value("SELECT COUNT(*) FROM `cache_logs_archived` WHERE `user_id`='&1'", 0, $this->nUserId)
-            + sql_value("SELECT COUNT(*) FROM `cache_reports` WHERE `userid`='&1'", 0, $this->nUserId)
-            == 0;
+        return !($login->userid != $this->nUserId && ($login->admin & ADMIN_USER) != ADMIN_USER);
```

Comment:
Das ist doch nicht das gleiche wie vorher? Auch wenn Du es jetzt einfach negiert hast. In dem zweiten return kann ja true und false bei rum kommen.